typing/typecore.ml

```diff
@@ -377,10 +377,10 @@ let unify_pat_types_return_equated_pairs ?(refine = None) loc env ty ty' =
 let unify_pat_types ?refine loc env ty ty' =
   ignore (unify_pat_types_return_equated_pairs ?refine loc env ty ty')
 
-let unify_pat ?refine env pat expected_ty =
+let unify_pat ?refine ?desc env pat expected_ty =
```

Comment:
A documentation comment and a more descriptive name would be nice. Maybe `sdesc_for_error`?

Reply:
Good idea. I added a comment and changed `desc` to `sdesc_for_hint`, that feels even more descriptive.